sky/serve/autoscalers.py

```diff
@@ -314,7 +314,19 @@ def select_outdated_replicas_to_scale_down(
             self,
             replica_infos: List['replica_managers.ReplicaInfo']) -> List[int]:
         """Select outdated replicas to scale down."""
+        all_replica_ids_to_scale_down = []
+        for info in replica_infos:
+            if info.version < self.latest_version and info.is_terminal:
+                all_replica_ids_to_scale_down.append(info.replica_id)
```

Comment:
IIRC the original behavior will terminate the failed replicas already. Is this only for removing the FAILED replica entries in the `serve status` table? In that case, should we make the `sky serve down` command able to remove a `FAILED` entry in the table instead?

I feel keeping the FAILED replicas for the previous version might be still useful.

Reply:
Yes, it is only for removing those failed records from a user's request. Agree that adding the ability to remove a record in `sky serve down` is a better idea. It is implemented in #3179 and I'll take a look soon 🫡